src/core/Target.mli

```diff
@@ -0,0 +1,144 @@
+(* Cooper Pierce
+ *
+ * Copyright (c) Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** Types for describing targets.
+
+    See also {!Input_to_core_t}, which has a similar set of types used when
+    pysemgrep generates targets that have slightly less information (e.g.,
+    these types have expanded information about the targets' locations). *)
+
+(** The location of a Semgrep target. This contains all of the target-specific
+    details needed to be able to determine how to scan a target, e.g.,
+
+    {ul
+      {- What products should we select rules from?}
+      {- Where can we get the contents of the target?}
+      {- What language should we analyze the target as?}
+    }
+
+    However, it does not contain the actual contents (parsed or otherwise) of
+    the target itself.
+ *)
+type t = Code of code | Lockfile of lockfile [@@deriving show]
```

Comment:
Code is a very bad name, as this data structure does not contain code.
Maybe call it RegularTarget,

Reply:
This is called `Code` so that it is consistent with the types @mmcqd introduced in https://github.com/semgrep/semgrep/pull/9695. I can change it here, but it would then no longer be consistent with the similar type added there.

I can rename both, and update the python code which generates this, but I am uncertain that this PR is the appropriate place to do so.

Reply:
I'm happy to change the name. Just wanted to indicate it's a target with source code in it. `RegularTarget` seems good to me. If Cooper is ok with changing the name here that's fine with me, or I can I open a separate PR after this one renaming everything.

Reply:
I have renamed this `Regular`, without the Target, (given it is in the module name already) in e18e0a6b4e5b7776a80ac46ff158b1490806f156.